packages/arb-avm-cpp/avm/include/avm/machine.hpp

```diff
@@ -57,10 +71,9 @@ class Machine {
 
     TuplePool& getPool() { return *machine_state.pool; }
 
-    SaveResults checkpoint(CheckpointStorage& storage);
-    bool restoreCheckpoint(const CheckpointStorage& storage,
-                           const std::vector<unsigned char>& checkpoint_key);
-    DeleteResults deleteCheckpoint(CheckpointStorage& storage);
+    void marshal_value(const value& val, std::vector<unsigned char>& buf) {
```

Comment:
curious why marshal_value is in the machine interface? seems like it you should be able to do it without a machine

Reply:
That's basically a compromise I made. The value type has CodePointRef whereas the marshal function produces the marshaling of a CodePoint. I put marshal value in the machine interface since it's linked to the code of the machine. In the future we can examine whether or not we could propagate the CodePointRef idea to the marshaling format as well.

This is a super thin wrapper around the global marshal_value function since it just calls `::marshal_value(val, buf, machine_state.static_values->code)`